arroyo-types/src/functions/hash.rs

```diff
@@ -0,0 +1,45 @@
+use base64::{engine::general_purpose::STANDARD_NO_PAD as encoder, Engine};
+use md5::{Digest, Md5};
+
+// would like to make this parameterized based on the hasher, since everything else looks the same
+// but I couldn't quite figure out what the common trait was between all of them...
+// I'll ask later if anyone has ideas.
+pub fn md5(argument: String) -> String {
+    let mut hasher = Md5::new();
+    hasher.update(argument.as_bytes());
+    let res = hasher.finalize();
+    // chose to encode as base64 to encode bytes to string, but not sure if that's correct
```

Comment:
Is `base64` str encoding the right choice here? utf-8 didn't make sense to me because it's not guaranteed that the hash is going to be a utf char.

Reply:
I answered this in-line, but it seems like md5 should be a hex-coded String and the others should return a DataType::Binary, which I'd presume will be a Vec<u8>.

Reply:
Yep, looks like the argument type for sha functions is also not right according to postgres docs.  DataType::Binary should be the argument and the return, I believe.



sha256(bytea) | bytea | SHA-256 hash | sha256('abc') | \xba7816bf8f01cfea414140de5dae2223​b00361a396177a9cb410ff61f20015ad
-- | -- | -- | -- | --

Reply:
Adjusted